File: DataBindingStudy/app/src/main/java/com/jie/databinding/BindingAdapterConversionActivity.java
package com.jie.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.jie.databinding.databinding.ActivityBindingAdapterConversionBinding;
import com.jie.databinding.model.Image;

import java.util.Random;


public class BindingAdapterConversionActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityBindingAdapterConversionBinding adapterBinding = DataBindingUtil.setContentView(this, R.layout.activity_binding_adapter_conversion);
        adapterBinding.setImageInfo(new Image("baidu"));
        adapterBinding.setPresenter(new Presenter());
        setTitle("BindingAdapter");
    }

    public class Presenter {
        public boolean onClick(Image image) {
            image.getUrl().set("xxxJackie" + new Random().nextInt(1000));
            return true;
        }
    }
}

File: DataBindingStudy/app/src/main/java/com/jie/databinding/ui/activity/BindingAdapterConversionActivity.kt
package com.jie.databinding.ui.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.jie.databinding.R
import com.jie.databinding.databinding.ActivityBindingAdapterConversionBinding
import com.jie.databinding.model.ImageEntity

class BindingAdapterConversionActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DataBindingUtil.setContentView<ActivityBindingAdapterConversionBinding>(
            this,
            R.layout.activity_binding_adapter_conversion
        ).apply {
            imageInfo = ImageEntity("baidu")
            presenter = Presenter()
        }
        title = "BindingAdapter study"
    }

    inner class Presenter {
        fun onClick(image: ImageEntity): Boolean {
            image.urlField?.set("xxxJackie ${(0..1000).random()}")
            return true
        }
    }
}